internal/action/command.go

```diff
@@ -977,6 +979,17 @@ func (h *BufPane) ReplaceCmd(args []string) {
 				return
 			}
 
+			if prevMatchEnd == locs[1] {
+				// skip empty match right after previous match
+				if searchLoc == end {
+					searchLoc = start
+				} else {
+					searchLoc = searchLoc.Move(1, h.Buf)
+				}
```

Comment:
...Also, this addresses the issue for `replace` but not for `find`. I.e.: we press Ctrl-f and type `$`, the cursor correctly moves to the end of the line, then we press Ctrl-n to find the next match but the cursor stays where it is, rather than moves to the end of the next line.

So shouldn't we move this logic into `h.Buf.FindNext()` instead?

Reply:
Also a side note: matching `^` (beginning of the line) still works not quite correctly (for a different reason). Looks like we should handle patterns beginning with `^` in a special way.

Reply:
> Why not do this just immediately after we found and replaced an empty match

That's not the same. For example, saying `replace '(z|$)' '@'` for
```
abc
xyz
```
would give
```
abc@
xy@@
```
with your approach, but
```
abc@
xy@
```
with `replaceall` and my variant. Of course, one can implement anything one wants, but I think it would be nice if `replace` with answer `y` all the give (until the search wraps over) gives the same result as `replaceall`.

> this addresses the issue for `replace` but not for `find`

That's true.

> So shouldn't we move this logic into `h.Buf.FindNext()` instead?

`FindNext` itself cannot move the cursor forward. How would `FindNext` know if an empty match is acceptable? Would that be another argument?

> matching `^` (beginning of the line) still works not quite correctly 

Can you share an example?

Reply:
> That's not the same. For example, saying `replace '(z|$)' '@'` for [...]

Indeed. Yeah, your version seems better then.



> `FindNext` itself cannot move the cursor forward. How would `FindNext` know if an empty match is acceptable? Would that be another argument?

Yep, I hadn't thought about the details, I was hoping we could generalize it between "find" and "replace" cases but seems like we need to address them separately...

> > matching `^` (beginning of the line) still works not quite correctly
> 
> Can you share an example?

Just `replace ^ @`, for example. Or e.g. `replace ^foo @` when the cursor is at `foo` but not at the beginning of a line.

Reply:
> I was hoping we could generalize it between "find" and "replace" cases but seems like we need to address them separately...

If we add a field like `LastMatch` to `Buffer` (besides `LastSearch`), then `h.Buf.FindNext()` could take this into account. So I now think it could work to make essentially all changes only once in `h.Buf.FindNext()`. At least it's worth a try.

Reply:
I don't think anymore that it's a good idea to modify `h.Buf.FindNext()`. That function may be called from scripts that shouldn't modify the search status of the buffer. One might be able to reuse a modified `h.FindNext()` in `ReplaceCmd`. Maybe it's better to do the necessary changes in `h.FindNext()` first and then see for an overlap.

I've just force-pushed a new version where I've just renamed `prevMatchEnd` to `lastMatchEnd` to align it with the naming of variables like `LastSearch`. Otherwise this PR seems fine to me provided that one only wants to modify `ReplaceCmd`.